### PR TITLE
Another attack of tidiness

### PR DESCRIPTION
**Pull request by mrks**
This fixes a couple of things that clang-tidy complains about. Mostly, it is functions that take a `shared_ptr` by copy. These are changed to `const std::shared_ptr&`.

This does not yet introduce any new error types.

@mrzzzrm - The merge conflicts introduced by this PR should be minimal. If not, feel free to overwrite those places with your version. clang-tidy will remind us to redo it.

### CODE REVIEW

File: src/lib/operators/insert.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ namespace opossum {
 // We need these classes to perform the dynamic cast into a templated ValueColumn
 class AbstractTypedColumnProcessor {
  public:
-  virtual ~AbstractTypedColumnProcessor() = default;
+  virtual ~AbstractTypedColumnProcessor() {}
```
Comment (mrzzzrm) [inline src/lib/operators/insert.cpp]: why does it prefer this over `=default`?

File: src/lib/operators/jit_operator/specialization/jit_code_specializer.cpp [hunk 2]
```diff
@@ -12,13 +12,13 @@
...
 #include <queue>
 
-#include "llvm_extensions.hpp"
 #include "jit_runtime_pointer.hpp"
+#include "llvm_extensions.hpp"
 
 namespace opossum {
 
 JitCodeSpecializer::JitCodeSpecializer(JitRepository& repository)
-    : _repository{repository}, _llvm_context{_repository.llvm_context()}, _compiler{_llvm_context} {}
```
Comment (mrzzzrm) [inline src/lib/operators/jit_operator/specialization/jit_code_specializer.cpp]: why was that removed?

File: src/test/operators/jit_operator/specialization/jit_compiler_test.cpp [hunk 3]
```diff
@@ -39,20 +39,20 @@ class JitCompilerTest : public BaseTest {
 };
 
 TEST_F(JitCompilerTest, CompilesAddedModules) {
-  auto compiler = JitCompiler(_context);
+  auto compiler = JitCompiler{};
```
Comment (mrzzzrm) [inline src/test/operators/jit_operator/specialization/jit_compiler_test.cpp]: why not pass the context anymore?

File: src/lib/optimizer/strategy/abstract_rule.hpp [hunk 4]
```diff
@@ -29,7 +29,7 @@ class AbstractRule {
    * IMPORTANT: Takes a copy of the node ptr because applying this rule to inputs of this node might remove this node
    * from the tree, which might result in this node being deleted if we don't take a copy of the shared_ptr here.
    */
-  bool _apply_to_inputs(std::shared_ptr<AbstractLQPNode> node);
+  bool _apply_to_inputs(const std::shared_ptr<AbstractLQPNode>& node);
```
Comment (mrzzzrm) [inline src/lib/optimizer/strategy/abstract_rule.hpp]: This was passed by value intentionally, see comment above.

**Comment by Bensk1** [comment]
This reminds me of the great discussion in #59

File: src/lib/operators/insert.cpp (hunk 1 above)
Comment (mrks) [reply src/lib/operators/insert.cpp]: It doesn't. My fault.

File: src/lib/operators/jit_operator/specialization/jit_code_specializer.cpp (hunk 2 above)
Comment (mrks) [reply src/lib/operators/jit_operator/specialization/jit_code_specializer.cpp]: The block displayed by Github is misleading. Only the compiler was removed because it is now default-constructible.

File: src/test/operators/jit_operator/specialization/jit_compiler_test.cpp (hunk 3 above)
Comment (mrks) [reply src/test/operators/jit_operator/specialization/jit_compiler_test.cpp]: Because the compiler doesn't use it.

File: src/lib/optimizer/strategy/abstract_rule.hpp (hunk 4 above)
Comment (mrks) [reply src/lib/optimizer/strategy/abstract_rule.hpp]: Oups, thanks.
